experimental/client.go

```diff
@@ -0,0 +1,338 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+
+	yaml "gopkg.in/yaml.v3"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+
+	"github.com/stolostron/go-template-utils/v3/pkg/templates"
+)
+
+func main() {
+	klog.InitFlags(nil)
+
+	var yamlFile, hubKubeConfigPath, clusterName string
+
+	flag.StringVar(&yamlFile, "i", "", "the input YAML file to resolve templates with")
+	flag.StringVar(&hubKubeConfigPath, "hub-kubeconfig", "", "the input kubeconfig to also resolve hub templates")
+	flag.StringVar(&clusterName, "cluster-name", "", "the cluster name to use to resolve hub templates")
```

Comment:
```suggestion
	flag.StringVar(&clusterName, "cluster-name", "", "the cluster name to use to resolve ManagedClusterName in hub templates")
```

Reply:
I used some slightly different wording. Let me know what you think.

Reply:
Looks good--thanks!